models/complaints.js

```diff
@@ -2,24 +2,27 @@ var mongoose = require('mongoose');
 var User = require('./user');
 var complaintSchema = new mongoose.Schema(
     {
-        name: { type: String, required: true },
-        desc: { type: String },
+        name: String,
+        desc: String,
+        createdAt: { type: Date, default: Date.now },
         reviewStartedAt: { type: Date },
+        reviewPendingAt: { type: Date },
+        reviewReopenAt: { type: Date },
+        reviewClosedAt: { type: Date },
         author: {
-            type: mongoose.Schema.Types.ObjectId,
-            ref: 'User',
+            id: {
+                type: mongoose.Schema.Types.ObjectId,
+                ref: 'User',
+            },
         },
         archived: { type: Boolean, default: false },
         archivingTime: { type: Date },
         status: {
             type: String,
-        } /** Pending => createdAt
-        Open => reviewStartedAt
-        Close => updatedAt */,
-        assignedTo: {
```

Comment:
assinedTo is supposed to be user object id ref to it. admin user id. not this string. in future it will be hard to add user id. when you see why string approach doesn't work